src/core/crypto/singleKey.ts

```diff
@@ -0,0 +1,194 @@
+import { Deserializer, Serializer } from "../../bcs";
+import {
+  AnyPublicKeyVariant,
+  AnySignatureVariant,
+  SigningScheme as AuthenticationKeyScheme,
+  SigningSchemeInput,
+} from "../../types";
+import { AuthenticationKey } from "../authenticationKey";
+import { Ed25519PublicKey, Ed25519Signature } from "./ed25519";
+import { AccountPublicKey, CryptoPublicKey, VerifyAccountSignatureArgs } from "./publicKey";
+import { Secp256k1PublicKey, Secp256k1Signature } from "./secp256k1";
+import { AccountSignature, CryptoSignature } from "./signature";
+
+/**
+ * Represents any public key supported by Aptos.
+ *
+ * Since [AIP-55](https://github.com/aptos-foundation/AIPs/pull/263) Aptos supports
+ * `Legacy` and `Unified` authentication keys.
+ *
+ * Any unified authentication key is represented in the SDK as `AnyPublicKey`.
+ */
+export class AnyPublicKey extends AccountPublicKey {
+  /**
+   * Reference to the inner public key
+   */
+  public readonly publicKey: CryptoPublicKey;
+
+  /**
+   * Index of the underlying enum variant
+   */
+  public readonly variant: AnyPublicKeyVariant;
+
+  // region Constructors
+
+  constructor(publicKey: CryptoPublicKey) {
+    super();
+    this.publicKey = publicKey;
+    if (publicKey.signatureScheme === SigningSchemeInput.Ed25519) {
+      this.variant = AnyPublicKeyVariant.Ed25519;
+    } else if (publicKey.signatureScheme === SigningSchemeInput.Secp256k1Ecdsa) {
+      this.variant = AnyPublicKeyVariant.Secp256k1;
+    } else {
+      throw new Error("Unsupported public key type");
+    }
+  }
+
+  // endregion
+
+  // region AccountPublicKey
+
+  /**
+   * Authentication scheme
+   */
+  public readonly authenticationScheme = AuthenticationKeyScheme.SingleKey;
+
+  verifySignature(args: VerifyAccountSignatureArgs): boolean {
+    const { message, signature } = args;
+    if (!(signature instanceof AnySignature)) {
+      return false;
+    }
+
+    return this.publicKey.verifySignature({
+      message,
+      signature: signature.signature,
+    });
+  }
+
+  // Note: need to explicitly set this overload, as the input used is different from the result of `toUint8Array`
+  authKey(): AuthenticationKey {
+    return AuthenticationKey.fromSchemeAndBytes({
+      scheme: this.authenticationScheme,
+      input: this.bcsToBytes(),
+    });
+  }
+
+  toUint8Array() {
+    // TODO (gabi): should this be the BCS-serialized key instead?
+    return this.publicKey.toUint8Array();
+  }
```

Comment:
I think we should use BCS-serialization for all of the `toUint8Array()` functions to be consistent. And then we can add a `toPublicKeyBytes()` as well. Wdy think?

Reply:
we already have `bcsToBytes`, so if `toUint8Array` and `bcsToBytes` are equivalent I would just drop `toUint8Array`

If you're suggesting a renaming of `toUint8Array` in something more explanatory, that could work.
Can be addressed in another PR for renaming

Reply:
I think in general `toUint8Array` and `bcsToBytes` give different results, since the bcs serialization is prefixed with the key length.

The question is.. what is the "bytes" representation of `AnyPublicKey`? Is it the same as the underlying crypto public key? Or should it be different?

I think it should be the former, which also plays well with authentication key derivation.
All instances of `AccountPublicKey` can then derive the associated authentication key with `AuthenticationKey.fromSchemeAndBytes(this.authenticationScheme, this.toUint8Array())`.

Reply:
Either works for me, just wanted it to be consistent. I see we also have a `serialize` function on some classes which may or may not do `bcsToBytes()`? I'm not too sure. Anyways I trust you :)

Reply:
fwiw, i think `toUint8Array` is more familiar and understandable in our dev community.

I think the bytes representation of  `AnyPublicKey` and `CryptoPublicKey` is different in the rust implementation level - but i might be wrong.

Reply:
>The question is.. what is the "bytes" representation of AnyPublicKey? Is it the same as the underlying crypto public key? Or should it be different?

@0xmaayan do you have an opinion?
Should I check in `aptos-core` to see what they do?

Reply:
as mentioned in my previous comment

I think the bytes representation of `AnyPublicKey` and `CryptoPublicKey` is different in the rust (aptos-core) implementation level - but i might be wrong, need to varify

Reply:
What I meant is.. which would you **personally** choose given your understanding.

Reply:
I went ahead and looked at the Rust code.
The `to_bytes` function for `AnyPublicKey` returns the bcs serialized value.
There's no `to_bytes` function for `AnySignature`.

So I'm updating the PR accordingly